controllers/villains.js

```diff
@@ -0,0 +1,33 @@
+const models = require('../models')
+
+const getAllVillains = async (request, response) => {
+  const villains = await models.villains.findAll()
+
+  return response.send(villains)
+}
+
+const getVillainBySlug = async (request, response) => {
+  const { slug } = request.params
+
+  const foundVillain = await models.villains.findOne({ where: { slug } })
+
+  if (!foundVillain) return response.status(404).send('404 Error')
+
+  return response.send(foundVillain)
+}
+
+const saveNewVillain = async (request, response) => {
+  const { name, realname, firstappearance, slug } = request.body
+
+  if (!name || !realname || !firstappearance || !slug) {
+    return response.status(400).send('The following fields are reauired: name, realname, firstappearnace, slug')
+  }
+
+  const newVillain = await models.villains.create({
+    name, realname, firstappearance, slug
```

Comment:
where is 'realname' saved? is this the movie?

i think thats the first appearance... 

did you test you create? i dont think that this is working... you are missing movie in your object and also you have two new variables called realname and first appearance which the DB or sequelize doesnt know about.